src/export/CodeSystemExporter.ts

```diff
@@ -55,15 +55,24 @@ export class CodeSystemExporter {
     }
   }
 
-  private updateCount(codeSystem: CodeSystem): void {
-    // Only update the count when it isn't explicitly set, when the content is 'complete',
-    // and there is at least one concept.
-    if (
-      codeSystem.count == null &&
-      codeSystem.content === 'complete' &&
-      codeSystem.concept?.length
-    ) {
-      codeSystem.count = codeSystem.concept.length;
+  private updateCount(codeSystem: CodeSystem, fshDefinition: FshCodeSystem): void {
+    // We can only derive a true count if the content is #complete
+    if (codeSystem.content === 'complete') {
+      const actualCount = codeSystem.concept?.length;
+      if (codeSystem.count == null && actualCount != null) {
+        codeSystem.count = actualCount;
+      } else if (codeSystem.count !== actualCount) {
+        const countRule = fshDefinition.rules.find(
+          r => r instanceof CaretValueRule && r.caretPath === 'count'
+        );
+        const sourceInfo = countRule?.sourceInfo ?? fshDefinition.sourceInfo;
+        logger.warn(
+          `The user-specified ^count (${codeSystem.count}) does not match the specified number of concepts ` +
+            `(${actualCount}). If this is not a "complete" ValueSet, set the ^content property to the appropriate ` +
```

Comment:
Two comments on this line:
1. If they specify a count but no concepts, then `actualCount` could be undefined here, so I think if it is undefined you want to default to printing `0`, not `undefined`.
2. Shouldn't "ValueSet" be "CodeSystem"?

Reply:
1. Good point.  I didn't want to default `undefined` `actualCount` to `0` above because I want `undefined` `actualAccount` to `===` `undefined` `count`.  But when I print it, I _should_ default it to `0`.  Thanks.
2. Ugh.

Reply:
1176b64.  There.  Happy now? Or would you like to point out any other embarrassingly obvious issues with this PR? ;-)

Reply:
I would've loved to, but I couldn't find any 😞